Sources/StreamChatUI/ChatMessageList/ChatMessageListVC+DiffKit.swift

```diff
@@ -8,12 +8,22 @@ import StreamChat
 extension ChatMessageListVC {
     /// Set the previous message snapshot before the data controller reports new messages.
     internal func setPreviousMessagesSnapshot(_ messages: [ChatMessage]) {
-        listView.previousMessagesSnapshot = messages
+        if let dataSourceDecorator {
+            listView.previousMessagesSnapshot = dataSourceDecorator(messages)
+        } else {
+            listView.previousMessagesSnapshot = messages
+        }
     }
 
     /// Set the new message snapshot reported by the data controller.
     internal func setNewMessagesSnapshot(_ messages: LazyCachedMapCollection<ChatMessage>) {
-        listView.currentMessagesFromDataSource = messages
-        listView.newMessagesSnapshot = messages
+        if let dataSourceDecorator {
+            let filteredMessages = dataSourceDecorator(Array(messages))
```

Comment:
the benchmark tests are failing, maybe good to run them again.

Reply:
Yes, that is the plan, thats why I'm adding a default decorator

Reply:
That line should be fine since we have background mapping enabled and if it is enabled, LazyCachedMapCollection maps [all the items in the init](https://github.com/GetStream/stream-chat-swift/blob/db65b60a2f79b2411f2f1d20aeae0ac1340bf70e/Sources/StreamChat/Utils/LazyCachedMapCollection.swift#L62). No performance hit here.